### PR TITLE
CompTaskInfo persistance

### PR DESCRIPTION
**Pull request by jiivan**
Whole CompTaskKeeper state is pickled in datadir.

### CODE REVIEW

File: golem/task/taskkeeper.py [hunk 1]
```diff
@@ -44,13 +48,41 @@ class CompTaskKeeper(object):
...
+    def restore(self):
+        logger.debug('COMPTASK RESTORE: %s', self.dump_path)
+        if not self.dump_path.exists():
+            logger.debug('No previous comptask dump found.')
+            return
+        with self.dump_path.open('rb') as f:
+            try:
+                active_tasks, subtask_to_task = pickle.load(f)
+            except (pickle.UnpicklingError, EOFError):
+                logger.exception('Problem restoring dumpfile: %s', self.dump_path)
```
Comment (badb) [inline golem/task/taskkeeper.py:84]: Maybe we should remove files that we cannot unpickle?

File: golem/task/taskkeeper.py [hunk 2]
```diff
@@ -114,9 +141,9 @@ class TaskHeaderKeeper(object):
 
     def __init__(self, environments_manager, min_price=0.0, app_version=APP_VERSION, remove_task_timeout=180,
                  verification_timeout=3600):
-        self.task_headers = {}  # all computing tasks that this node now about
+        self.task_headers = {}  # all computing tasks that this node nows about
```
Comment (badb) [inline golem/task/taskkeeper.py]: knows

File: tests/golem/task/test_taskkeeper.py [hunk 3]
```diff
@@ -223,9 +230,36 @@ def test_init(self):
         self.assertIsInstance(csi, CompSubtaskInfo)
 
 
-class TestCompTaskKeeper(LogTestCase):
-    def test_comp_keeper(self):
-        ctk = CompTaskKeeper()
+class TestCompTaskKeeper(LogTestCase, PEP8MixIn):
```
Comment (badb) [inline tests/golem/task/test_taskkeeper.py]: Can you use `golem.testutils.TempDirFixture` here?

File: tests/golem/task/test_taskkeeper.py [hunk 4]
```diff
@@ -223,9 +230,36 @@ def test_init(self):
...
+    PEP8_FILES = [
+        "golem/task/taskkeeper.py",
+    ]
+    def setUp(self):
+        super(TestCompTaskKeeper, self).setUp()
+        random.seed()
+
+    def test_persistance(self):
+        """Tests wether tasks are persistent between restarts."""
```
Comment (badb) [inline tests/golem/task/test_taskkeeper.py]: `whether`?

**Review by badb — CHANGES_REQUESTED** [review]
Additionally I've been getting an error on `test_conformance`. How can I check why?

File: golem/task/taskkeeper.py (hunk 1 above)
Comment (jiivan) [reply golem/task/taskkeeper.py:84]: I'm not sure about this. Anyway this file will be overwritten in next dump().

File: golem/task/taskkeeper.py (hunk 2 above)
Comment (jiivan) [reply golem/task/taskkeeper.py]: :heavy_check_mark:

File: tests/golem/task/test_taskkeeper.py (hunk 3 above)
Comment (jiivan) [reply tests/golem/task/test_taskkeeper.py]: :heavy_check_mark:

File: tests/golem/task/test_taskkeeper.py (hunk 4 above)
Comment (jiivan) [reply tests/golem/task/test_taskkeeper.py]: :heavy_check_mark:

**Comment by jiivan** [comment]
Hmm, they're written to stdout. Here it looks like this (my underline):
```
(golem) dybi@labelle:~/repo/golem$ nosetests tests/golem/task/test_taskkeeper.py
..F.........
======================================================================
FAIL: Test that we conform to PEP-8.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dybi/repo/golem/golem/testutils.py", line 136, in test_conformance
    self.assertEqual(result.total_errors, 0, "Found code style errors (and warnings).")
AssertionError: Found code style errors (and warnings).
-------------------- >> begin captured stdout << ---------------------
golem/task/taskkeeper.py:17:52: E261 at least two spaces before inline comment
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 12 tests in 1.676s

FAILED (failures=1)
```

**Review by badb — APPROVED** [review]

**Pull request merged**